tests/foreman/ui/test_search.py

```diff
@@ -0,0 +1,213 @@
+"""Test classes for Search component
+
+:Requirement: Search
+
+:CaseAutomation: Automated
+
+:CaseComponent: Search
+
+:Team: Endeavour
+
+:CaseImportance: Medium
+"""
+from collections import namedtuple
+
+from fauxfactory import gen_string
+import pytest
+
+from robottelo.config import settings
+
+SearchData = namedtuple('SearchData', ['expected_items', 'not_expected_items'])
+
+
+def menu_search_should_find(search_func: callable, values: list[str]):
+    """Helper function that does a vertical menu search for each value in `values` list
+    and makes sure it finds a corresponding match.
+    """
+    for search_string in values:
+        results = search_func(search_string)
+        assert results, f'No result found for search string {search_string}!'
+        for result in results:
+            assert (
+                search_string.lower() in result.lower()
+            ), f'Search string {search_string} does not match result {result}!'
+
+
+def menu_search_should_not_find(search_func: callable, values: list[str]):
```

Comment:
not a blocker but I'm curious if running the above function with `not` in front of it would achieve the same result?

Reply:
I'd agree, there's probably a way to do this without a separate helper. Doesn't really affect the test overall though, more of a style difference.

Reply:
I get that the first three lines of the two functions are pretty similar, but merging both functions together to one and adding some "bool" argument, i.e., expect_to_find=True/False, would lead to more if..else branching and thus to less readability, IMO. Esp. comparing to how many lines of code we save by that.

Using `not menu_search_should_find()` here won't help, because `menu_search_should_find()` doesn't return bool, but does asserts instead - so a test failure is easily identifiable in the results.
Doing something like this
```
with pytest.raises(AssertionError):
    menu_search_should_find()
```
would solve it, but then there is a case that AssertionError is thrown by the second assert (line 31) and that would be wrong.